src/Compilers/Shared/DesktopBuildClient.cs

```diff
@@ -292,7 +292,35 @@ protected override int RunLocalCompilation(string[] arguments, BuildPaths buildP
                 cancellationToken.ThrowIfCancellationRequested();
 
                 Log("Attempt to connect named pipe '{0}'", pipeName);
-                pipeStream.Connect(timeoutMs);
+
+                // .NET 4.5 implementation of NamedPipeStream.Connect busy waits the entire time.
+                // Work around is to spin wait.
+                const int maxWaitInterval = 50;
+                int elapsed = 0;
+                var sw = new SpinWait();
+                while (true)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    int waitTime = Math.Min(timeoutMs - elapsed, maxWaitInterval);
+                    if (timeoutMs != Timeout.Infinite && waitTime <= 0)
+                    {
+                        Log($"Connecting to server timed out after {timeoutMs} ms");
+                        break;
```

Comment:
@stephentoub No that should be a `return null;` instead of a `break` -- nice catch. I will add a test for this.